### PR TITLE
suggestions were made

### PR DESCRIPTION
**Pull request by Sree0076**

### CODE REVIEW

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts [hunk 1]
```diff
@@ -1,52 +1,63 @@
...
   templateUrl: './vendor-listing.component.html',
   styleUrl: './vendor-listing.component.scss',
 })
 export class VendorListingComponent implements OnInit {
- 
   vendors!: IVendor[];
 
-  constructor(private vendorService: VendorService) {}
-  
+  constructor(private vendorService: VendorService,private router : Router) {}
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.ts]: Space after comma

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts [hunk 2]
```diff
@@ -1,52 +1,63 @@
...
    });
- }
 }
+
+  /**
+   * Method to get the list of vendors 
+   */
+  navigateToView(event: IVendor)
+  {
+    console.log(event);
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.ts]: remove log

File: src/app/features/vendor/vendor.module.ts [hunk 3]
```diff
@@ -17,12 +17,12 @@ import { FloatLabelModule } from "primeng/floatlabel"
...
-import { ConfirmationService } from 'primeng/api';
+import { ConfirmationService, MessageService } from 'primeng/api';
 
 
 @NgModule({
   declarations: [VendorListingComponent, VendorCreationComponent, VendorViewComponent],
-  imports: [CommonModule, SharedModule, VendorRoutingModule, TableModule, FormsModule, FloatLabelModule, ReactiveFormsModule, DropdownModule, ConfirmDialogModule, ToastModule, ButtonModule, MultiSelectModule, InputTextModule, InputGroupModule, InputGroupAddonModule],
-  providers: [ConfirmationService],
+  imports: [CommonModule, SharedModule, VendorRoutingModule, TableModule,ToastModule, FormsModule, FloatLabelModule, ReactiveFormsModule, DropdownModule, ConfirmDialogModule, ToastModule, ButtonModule, MultiSelectModule, InputTextModule, InputGroupModule, InputGroupAddonModule],
+  providers: [ConfirmationService,MessageService],
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor.module.ts]: space after comma

File: src/app/features/vendor/vendor-listing/vendor-listing.component.scss [hunk 4]
```diff
@@ -6,6 +6,11 @@
     /* Space for the sidebar */
 }
 
+.header {
+    height: 3rem;
+    border: 0.063rem solid #d1d5db;
+    background-color: #1a4870;
+  }
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.scss]: line gap after this

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 5]
```diff
@@ -1,3 +1,5 @@
+
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: remove this line

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 6]
```diff
@@ -1,3 +1,5 @@
...
+import { Message } from './../../../../../node_modules/esbuild/lib/main.d';
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: do we need this?

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 7]
```diff
@@ -62,12 +65,14 @@ export class VendorCreationComponent implements OnInit {
   markets!: IDropDownFields[];
   services!: IDropDownFields[];
   selectedVendorId!: number;
-
+  vendorName='';
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: do we need this varibale

File: src/app/features/vendor/vendor-creation/vendor-creation.component.ts [hunk 8]
```diff
@@ -86,7 +91,7 @@ export class VendorCreationComponent implements OnInit {
    */
   createAddVendorForm(): void {
     this.addVendorForm = this.fb.group({
-      vendorName: ['', [Validators.required, Validators.maxLength(100)]],
+      vendorName: [{value:'',disabled: this.isEdit }, [Validators.required, Validators.maxLength(100)]],
```
Comment (athul-sanjeevan) [inline src/app/features/vendor/vendor-creation/vendor-creation.component.ts]: space after comma

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts (hunk 1 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.ts]: ok

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts (hunk 2 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.ts]: done

File: src/app/features/vendor/vendor.module.ts (hunk 3 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor.module.ts]: ok

File: src/app/features/vendor/vendor-listing/vendor-listing.component.scss (hunk 4 above)
Comment (Sree0076) [reply src/app/features/vendor/vendor-listing/vendor-listing.component.scss]: done

File: src/app/features/vendor/vendor-listing/vendor-listing.component.ts [hunk 9]
```diff
@@ -31,22 +26,30 @@ import { VendorService } from '../services/vendor.service';
...
     this.vendors = vendors;
     this.vendors.sort((a, b) => a.name.localeCompare(b.name));
    });
- }
 }
+
+  /**
+   * Method to get the list of vendors 
+   */
+  navigateToView(event: IVendor)
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-listing/vendor-listing.component.ts]: return type missing. Curly braces should be in same line of method name

File: src/app/features/vendor/vendor-view/vendor-view.component.ts [hunk 10]
```diff
@@ -80,9 +81,9 @@ export class VendorViewComponent {
   /**
    * Method to approve the vendor by updating their status to approved.
    */
-  approveVendor(): void {
-    this.vendorService.approveVendor(this.vendor.id).subscribe(() => {
-      this.getVendorById(this.vendor.id);
+   approveVendor(): void {
+    this.vendorService.approveVendor(this.vendor.id).subscribe( () => {
+       this.showSuccess("Vendor Approves Successfully");
```
Comment (kavyamolr) [inline src/app/features/vendor/vendor-view/vendor-view.component.ts]: Approved

**Review by kavyamolr — APPROVED** [review]

**Review by athul-sanjeevan — APPROVED** [review]

**Pull request merged**